### PR TITLE
Fix pipeline module evaluation when contiguous activation checkpoin…

### PR DESCRIPTION
**Pull request by hablb**
When activation checkpointing with contiguous checkpoints are enabled calling consecutive eval_batch() will result in 

File "/deepspeed/runtime/activation_checkpointing/checkpointing.py", line 421, in partition_activations
    contiguous_data_buffers[i][data_offsets[i]].data[range(
IndexError: list index out of range

Must call deepspeed.runtime.activation_checkpointing.reset() after forward to reset internal buffers.
In training these buffers are reset in CheckpointFunction.backward()

### CODE REVIEW

**Comment by hablb** [comment]
@microsoft-github-policy-service agree company="Habana"

**Comment by hablb** [comment]
@tjruwase Can you please review?

File: tests/unit/pipe/test_pipe_module.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def sequential_model():
 @pytest.fixture
 def simple_config():
     config_dict = {
-        "train_batch_size": 1,
+        "train_batch_size": 2,
```
Comment (tjruwase) [inline tests/unit/pipe/test_pipe_module.py:36]: Why is this batch _size change needed?

**Review by tjruwase — APPROVED** [review]

**Comment by hablb** [comment]
@tjruwase Can you please approve running the workflows?

**Pull request merged**